subprojects/core/src/main/java/org/gradle/api/services/internal/RegisteredBuildServiceProvider.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.services.internal;
+
+import com.google.common.collect.ImmutableList;
+import org.gradle.api.artifacts.component.BuildIdentifier;
+import org.gradle.api.services.BuildService;
+import org.gradle.api.services.BuildServiceParameters;
+import org.gradle.internal.Try;
+import org.gradle.internal.instantiation.InstantiationScheme;
+import org.gradle.internal.isolated.IsolationScheme;
+import org.gradle.internal.isolation.IsolatableFactory;
+import org.gradle.internal.service.ServiceLookup;
+import org.gradle.internal.service.ServiceRegistry;
+
+import javax.annotation.Nullable;
+
+// TODO:configuration-cache - complain when used at configuration time, except when opted in to this
+@SuppressWarnings("rawtypes")
```

Comment:
💄 I'd reduce the suppression scope even if I had to annotate more than one location.

Reply:
Right, blindly kept it from the original BuildServiceProvider